lib/Backends/OpenCL/OpenCLDeviceManager.h

```diff
@@ -35,20 +35,39 @@ namespace runtime {
 class OpenCLBuffer {
   /// The OpenCL buffer being stored.
   cl_mem buffer_;
+
   /// Count of functions using this buffer.
   unsigned int users_{0};
+
   /// Size of the buffer in bytes.
   const size_t size_{0};
 
+  /// CL command queue for the network.
+  cl_command_queue commands_;
+
 public:
-  ~OpenCLBuffer() { clReleaseMemObject(buffer_); }
-  OpenCLBuffer(cl_mem &buffer, size_t size) : buffer_(buffer), size_(size) {}
+  ~OpenCLBuffer() {
+    clReleaseMemObject(buffer_);
+    clReleaseCommandQueue(commands_);
+  }
+
+  OpenCLBuffer(cl_mem buffer, size_t size) : buffer_(buffer), size_(size) {}
+
   /// Returns the stored buffer.
   cl_mem getBuffer() { return buffer_; }
+
+  /// Returns the command queue.
+  cl_command_queue getCommandQueue() { return commands_; }
+
+  /// Sets the command queue.
+  void setCommandQueue(cl_command_queue command) { commands_ = command; }
```

Comment:
I'm increasingly convinced setters are a bad idea.  Is it valid to have an OpenCLBuffer without a command queue?  If not, put it in the constructor.  If it is valid, why?